packages/edit-site/src/components/resizable-frame/index.js

```diff
@@ -154,13 +155,43 @@ function ResizableFrame( {
 		if ( remainingWidth > SNAP_TO_EDIT_CANVAS_MODE_THRESHOLD ) {
 			// Reset the initial aspect ratio if the frame is resized slightly
 			// above the sidebar but not far enough to trigger full screen.
-			setFrameSize( { width: '100%', height: '100%' } );
+			setFrameSize( INITIAL_FRAME_SIZE );
 		} else {
 			// Trigger full screen if the frame is resized far enough to the left.
 			setCanvasMode( 'edit' );
 		}
 	};
 
+	// Handle resize by arrow keys
+	const handleResizableHandleKeyDown = ( event ) => {
+		if ( ! [ 'ArrowLeft', 'ArrowRight' ].includes( event.key ) ) {
+			return;
+		}
+
+		const step = 20 * ( event.shiftKey ? 5 : 1 );
+		const delta = step * ( event.key === 'ArrowLeft' ? 1 : -1 );
+		const newWidth = Math.max(
+			FRAME_MIN_WIDTH,
+			frameRef.current.resizable.offsetWidth + delta
+		);
+
+		setFrameSize( {
+			width: newWidth,
+			height: calculateNewHeight(
+				newWidth,
+				initialAspectRatioRef.current
+			),
+		} );
+
+		// If oversized, immediately switch to edit mode
+		if ( newWidth > initialComputedWidthRef.current ) {
+			setFrameSize( INITIAL_FRAME_SIZE );
+			setCanvasMode( 'edit' );
+			// TODO: Confirm if this is the focus behavior we want
+			document.querySelector( 'iframe[name=editor-canvas]' ).focus();
```

Comment:
Needs accessibility feedback: Do we want some kind of focus manipulation after the frame is resized to full width? (cc @afercia @andrewhayward)

## To test

1. Focus on the resize handle.
2. Hit the left arrow key until the editor canvas is in full width (edit mode).
3. Hit the tab key to see where the focus is.

Comment out line 191 to see the default behavior.

Reply:
I think it would be good if focus behaved the same in this case as it does when moving to edit mode by pressing Enter after focusing the frame, where we land at the top of the editing canvas and Tab focuses the first block in Select mode. 

One thing I'm noticing is that focusing on resize handle and pressing left arrow once sometimes takes me into edit mode straightaway. I'm not sure if that's desirable, as currently the drag handle, when dragged with a mouse, allows us to expand the canvas until the sidebar is about 100px wide, and only if we try expanding further than that does it switch to edit mode. Dragging with a mouse is also behaving a bit weirdly in this branch:


https://github.com/WordPress/gutenberg/assets/8096000/3c6d2ca4-15d6-47b8-a427-ff221bfcf484

(I can't reproduce this 100% of the time, but it's happening consistently on Firefox and about half the time on Chrome)

Reply:
Hello @mirka @tellthemachines I'd agree focus management should work in the same way it works when switching to the Edit view when pressing Enter on the iframe. However, I'm not sure it should land on the first block... I created https://github.com/WordPress/gutenberg/issues/51570 a few days ago to reconsider initial focus on the Site editor > Edit view.